analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ReturnEmptyCollectionInsteadOfNull.CSharp8.cs

```diff
@@ -25,3 +25,13 @@ str switch
             };
     }
 }
+
+// Reproducer for #6878 https://github.com/SonarSource/sonar-dotnet/issues/6494
+namespace Issue_6878
+{
+    class MyClass
+    {
+        public byte[]? GetNullableArray(string itemId) => null; // Noncompliant FP, Should not raise because nullable is allowed
```

Comment:
Please add all combinations of nullabilities:
```suggestion
        public byte[]? GetNullableArrayOfNonNullableValueType(string itemId) => null; // Noncompliant FP, Should not raise because nullable is allowed
        public byte?[] GetNonNullableArrayOfNullableValueType(string itemId) => null; // Noncompliant
        public byte?[]? GetNullableArrayOfNullableValueType(string itemId) => null; // Noncompliant, FP
        
        public object[]? GetNullableArrayOfNonNullableReferenceType(string itemId) => null; // Noncompliant FP
        public object?[] GetNonNullableArrayOfNullableReferenceType(string itemId) => null; // Noncompliant
        public object?[]? GetNullableArrayOfNullableReferenceType(string itemId) => null; // Noncompliant FP
```